internal/integration/integration_test.go

```diff
@@ -619,24 +620,46 @@ func TestCancelCheckerDeletePod(t *testing.T) {
 	}
 	tc.AssertCancelled(ctx, build)
 
-	// Give it time to delete
-	time.Sleep(5 * time.Second)
-
 	// TriggerBuild performs this type assertion
 	job := build.Jobs.Edges[0].Node.(*api.JobJobTypeCommand)
 
+	jobName := cfg.JobPrefix + job.Uuid
+	opts := metav1.ListOptions{
+		LabelSelector: fields.OneTermEqualSelector("batch.kubernetes.io/job-name", jobName).String(),
+	}
+
+	// wait for pods to be deleted
+	if err := roko.NewRetrier(
+		roko.WithMaxAttempts(5),
+		roko.WithStrategy(roko.Exponential(2*time.Second, 0)),
+	).DoWithContext(ctx, func(r *roko.Retrier) error {
+		result, err := hasJobPod(tc, ctx, job)
+		if err != nil {
+			return err
+		}
+		if result {
+			return fmt.Errorf("found at least one pods, waiting for deletion")
+		}
+		return nil
+	}); err != nil {
+		t.Fatalf("kubernetes.CoreV1().Pods(%q).List(ctx, %v): pods were not deleted after retries. Final error: %v",
+			cfg.Namespace, opts, err)
+	}
+}
+
+func hasJobPod(tc testcase, ctx context.Context, job *api.JobJobTypeCommand) (bool, error) {
 	jobName := cfg.JobPrefix + job.Uuid
 	opts := metav1.ListOptions{
 		LabelSelector: fields.OneTermEqualSelector("batch.kubernetes.io/job-name", jobName).String(),
 	}
 	pods, err := tc.Kubernetes.CoreV1().Pods(cfg.Namespace).List(ctx, opts)
 	if err != nil {
-		t.Fatalf("kubernetes.CoreV1().Pods(%q).List(ctx, %v) error = %v", cfg.Namespace, opts, err)
+		return false, err
 	}
 	if len(pods.Items) > 0 {
-		t.Fatalf("kubernetes.CoreV1().Pods(%q).List(ctx, %v): found %d pods, there should be none (they should have been deleted). Pods:\n%s",
-			cfg.Namespace, opts, len(pods.Items), pretty.Sprint(pods.Items))
+		return true, nil
 	}
+	return false, nil
```

Comment:
Another thought I had reading this (take it or leave it):

```suggestion
return len(pods.Items) > 0, nil
```